lib/Gateway.js

```diff
@@ -1680,65 +1702,76 @@ Gateway.prototype.discoverValue = function (node, vId) {
+              )
               off = 23 // Closed state
               break
             case 'Cover status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.OPENING
+              )
               break
             case 'Door state':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.DOOR
+              )
               off = 1 // Door closed on payload 1
               break
+            case 'Alarm status':
             case 'Dust in device status':
+            case 'Load error status':
             case 'Over-current status':
             case 'Over-load status':
             case 'Hardware status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.PROBLEM
+              )
               break
             case 'Heat sensor status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.HEAT
+              )
               break
             case 'Motion sensor status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.MOTION
+              )
               break
-            case 'Alarm status':
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.CONNECTIVITY
+            case 'Water Alarm':
+              cfg = setBinarySensorPayload(
+                Constants.deviceClass.sensor_binary.MOISTURE
+              )
               break
             // sensor status has multiple Properties. therefore is good to work
             // on property basis... user friendly
             case 'Sensor status':
               switch (valueId.propertyName) {
                 case 'Smoke Alarm':
-                  cfg.discovery_payload.device_class =
+                  cfg = setBinarySensorPayload(
                     Constants.deviceClass.sensor_binary.SMOKE
+                  )
                   break
                 case 'Water Alarm':
-                  cfg.discovery_payload.device_class =
+                  cfg = setBinarySensorPayload(
                     Constants.deviceClass.sensor_binary.MOISTURE
+                  )
                   break
                 default:
               }
-              cfg.object_id = valueId.propertyName
-              break
-            case 'Water Alarm':
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.MOISTURE
+              discoveredObjectId = valueId.propertyName
               break
             default:
           }
+          // cfg not there?
+          cfg = cfg ? copy(cfg) : copy(hassCfg.binary_sensor)
```

Comment:
```suggestion
          cfg = cfg || copy(hassCfg.binary_sensor)
```

No need to copy it here, it's already copied in `getBinarySensorConfig` function